Orange/widgets/data/owsave.py

```diff
@@ -22,58 +39,108 @@ class Inputs:
     resizing_enabled = False
 
     last_dir = Setting("")
-    last_filter = Setting("")
     auto_save = Setting(False)
+    filetype = Setting(FILE_TYPES_NAMES[0])
+    compression = Setting(COMPRESSIONS_NAMES[0])
+    compress = Setting(False)
 
     @classmethod
     def get_writers(cls, sparse):
         return [f for f in FileFormat.formats
                 if getattr(f, 'write_file', None) and getattr(f, "EXTENSIONS", None)
                 and (not sparse or getattr(f, 'SUPPORT_SPARSE_DATA', False))]
 
+    def get_writer_selected(self):
+        type = FILE_TYPES[self.filetype]
+        compression = COMPRESSIONS[self.compression] if self.compress else ''
+        writer = FileFormat.get_reader(type)
+        writer.EXTENSIONS = [writer.EXTENSIONS[writer.EXTENSIONS.index(type + compression)]]
```

Comment:
You are looking up an index of something in the list and then indexing that list with the found index... you will probably be getting that "something" back most of the time ;)
Writing this directly `writer.EXTENSIONS = [type + compression]` looks a bit dangerous since you are overwriting the list, but it should always be one of the available options anyway. Probably it would not be bad to have an assert before this to make sure that the chosen extension was really supported (more for the benefit of someone reading the code).